clients/src/main/java/org/apache/kafka/common/serialization/Serialization.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license agreements. See the NOTICE
+ * file distributed with this work for additional information regarding copyright ownership. The ASF licenses this file
+ * to You under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package org.apache.kafka.common.serialization;
+
+/**
+ * The interface for wrapping a serializer and deserializer for the given data type.
+ *
+ * @param <T>
+ */
+public interface Serialization<T> {
+
+    Serializer<T> serializer();
```

Comment:
`Serializer` and `Deserializer` are existing Kafka concepts and they are used in a number of places. The currently proposed interface has the advantage that you can use existing serializers and deserializers, but it has the downside you suggested.

The alternative is to have a `SerDe` like you suggest and then have a method that given a `Serializer` and `Deserializer` returns a `SerDe` implementation. One question is: do we want `SerDe` to be in `o.a.kafka.common` even though it's not used anywhere but Kafka Streams?